src/ZodError.ts

```diff
@@ -217,12 +217,24 @@ export class ZodError<T = any> extends Error {
     this.issues = [...this.issues, ...subs];
   };
 
-  flatten = <U = string>(
+  flatten(
+    mapper?: (issue: ZodIssue) => string
+  ): {
+    formErrors: string[];
+    fieldErrors: { [k: string]: string[] };
+  };
+  flatten<U>(
+    mapper?: (issue: ZodIssue) => U
+  ): {
+    formErrors: U[];
+    fieldErrors: { [k: string]: U[] };
+  };
+  flatten<U = string>(
     mapper: (issue: ZodIssue) => U = (issue: ZodIssue) => issue.message as any
   ): {
     formErrors: U[];
     fieldErrors: { [k: string]: U[] };
-  } => {
+  } {
```

Comment:
I so rarely use this method syntax: are there any gotchas or considerations for changing from an arrow function bound to a property?

Reply:
The only difference is this case, is that it will technically allow user to change `this` context with `bind`, `call` and `apply`. But I'd say that is an edge-case.

Reply:
💯 . You're welcome. Agreed, it's always those nuances that can be hard to catch.